xlators/mgmt/glusterd/src/glusterd-utils.c

```diff
@@ -5491,8 +5495,12 @@ glusterd_compare_friend_data(dict_t *peer_data, int32_t *status, char *hostname)
         goto out;
     }
 
+    arg = GF_CALLOC(1, sizeof(*arg), gf_common_mt_char);
```

Comment:
We could allocate the space for the bitmap dynamically based on `count`, instead of using a fixed size array. You also need to check allocation errors.